### ISSUE TITLE
PushNotifications requestIOSPermissions not working in @latest (^3.0.11)

### ISSUE THREAD

**Issue by joebernard** [aws-amplify/amplify-js#5794]
This bug has returned: https://github.com/aws-amplify/amplify-js/issues/4574

**Describe the bug**
The push notifications configuration [`requestIOSPermissions`](https://aws-amplify.github.io/docs/js/push-notifications) seems to have stopped working. The app prompts for push notification permissions on app load even if this config is `false`.  If I switch to the `@unstable` release (^3.0.11-unstable.40) it is still broken. It did work in versions prior to 2.1.1.

This is a React Native project v0.62.2

The feature was introduced in #1781 

**To Reproduce**
`yarn add @aws-amplify/pushnotification@latest`

**Expected behavior**
Should respect `requestIOSPermissions` configuration.

<details>
  <summary><strong>Environment</strong></summary>

```
System:
    OS: macOS 10.15.4
    CPU: (12) x64 Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz
    Memory: 2.05 GB / 32.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 10.15.3 - ~/.nvm/versions/node/v10.15.3/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 6.14.5 - ~/.nvm/versions/node/v10.15.3/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Browsers:
    Chrome: 81.0.4044.138
    Firefox: 76.0
    Safari: 13.1
  npmPackages:
    @aws-amplify/analytics: ^3.1.10 => 3.1.10 
    @aws-amplify/auth: ^3.2.7 => 3.2.7 
    @aws-amplify/cache: ^3.1.10 => 3.1.10 
    @aws-amplify/core: ^3.2.7 => 3.2.7 
    @aws-amplify/pushnotification: ^3.0.11 => 3.0.11 
    @aws-amplify/storage: ^3.2.0 => 3.2.0 
    @babel/core: ^7.6.2 => 7.9.0 
    @babel/runtime: ^7.6.2 => 7.9.2 
    @react-native-community/async-storage: 1.10.1 => 1.10.1 
    @react-native-community/eslint-config: ^0.0.5 => 0.0.5 
    @react-native-community/masked-view: ^0.1.10 => 0.1.10 
    @react-native-community/netinfo: ^5.8.0 => 5.8.1 
    @react-native-community/picker: ^1.3.0 => 1.5.0 
    @react-native-community/push-notification-ios: 1.2.0 => 1.2.0 
    @types/react: 16.9.35 => 16.9.35 
    amazon-cognito-identity-js: 4.2.4 => 4.2.4 
    apollo-cache: ^1.3.5 => 1.3.5 
    apollo-cache-inmemory: ^1.6.6 => 1.6.6 
    apollo-client: 2.6.10 => 2.6.10 
    apollo-link: ^1.2.14 => 1.2.14 
    apollo-link-http: ^1.5.17 => 1.5.17 
    apollo-utilities: ^1.3.4 => 1.3.4 
    aws-amplify: ^3.0.11 => 3.0.11 
    aws-appsync: ^3.0.3 => 3.0.3 
    aws-appsync-auth-link: ^2.0.2 => 2.0.2 
    aws-appsync-react: ^3.0.3 => 3.0.3 
    aws-appsync-subscription-link: ^2.1.0 => 2.1.0 
    babel-jest: ^24.9.0 => 24.9.0 
    bugsnag-react-native: ^2.23.7 => 2.23.7 
    eslint: ^6.5.1 => 6.8.0 
    graphql: 15.0.0 => 15.0.0 
    graphql-tag: ^2.10.3 => 2.10.3 
    jest: ^24.9.0 => 24.9.0 
    lodash.clonedeep: ^4.5.0 => 4.5.0 
    lodash.debounce: ^4.0.8 => 4.0.8 
    lodash.flowright: ^3.5.0 => 3.5.0 
    lodash.uniqby: ^4.7.0 => 4.7.0 
    metro-react-native-babel-preset: ^0.58.0 => 0.58.0 
    moment: ^2.25.3 => 2.25.3 
    moment-timezone: ^0.5.28 => 0.5.28 
    native-base: ^2.13.12 => 2.13.12 
    prop-types: ^15.7.2 => 15.7.2 
    react: 16.11.0 => 16.11.0 
    react-apollo: ^3.1.5 => 3.1.5 
    react-dom: ^16.12.0 => 16.13.1 
    react-native: 0.62.2 => 0.62.2 
    react-native-animatable: ^1.3.3 => 1.3.3 
    react-native-camera: ^3.24.1 => 3.26.0 
    react-native-circular-progress: 1.3.6 => 1.3.6 
    react-native-config: 1.0.0 => 1.0.0 
    react-native-country-picker-modal: ^1.10.0 => 1.10.0 
    react-native-device-info: ^5.5.7 => 5.5.7 
    react-native-fast-image: ^8.1.5 => 8.1.5 
    react-native-fs: ^2.16.6 => 2.16.6 
    react-native-gesture-handler: ^1.6.1 => 1.6.1 
    react-native-haptic-feedback: ^1.9.0 => 1.10.0 
    react-native-orientation-locker: ^1.1.8 => 1.1.8 
    react-native-permissions: ^2.1.4 => 2.1.4 
    react-native-progress: ^4.1.2 => 4.1.2 
    react-native-reanimated: ^1.8.0 => 1.8.0 
    react-native-safe-area-context: ^1.0.0 => 1.0.0 
    react-native-safe-area-view: ^1.1.1 => 1.1.1 
    react-native-screens: ^2.7.0 => 2.7.0 
    react-native-share: 3.3.2 => 3.3.2 
    react-native-svg: ^12.1.0 => 12.1.0 
    react-native-svg-animated-linear-gradient: https://github.com/joebernard/react-native-svg-animated-linear-gradient.git => 0.3.4 
    react-native-tab-view: ^2.14.0 => 2.14.0 
    react-native-vector-icons: ^6.6.0 => 6.6.0 
    react-native-video: ^5.0.2 => 5.0.2 
    react-native-webview: 9.4.0 => 9.4.0 
    react-navigation: ^4.3.9 => 4.3.9 
    react-navigation-props-mapper: ^1.0.4 => 1.0.4 
    react-navigation-stack: ^2.4.0 => 2.5.0 
    react-navigation-tabs: ^2.8.13 => 2.8.13 
    react-test-renderer: 16.11.0 => 16.11.0 
    uuid: ^3.4.0 => 3.4.0 
  npmGlobalPackages:
    @aws-amplify/cli: 4.18.1
    gatsby-cli: 2.12.10
    ios-deploy: 1.10.0
    npm: 6.14.5
    react-native-create-library: 3.1.2
    serverless: 1.70.1

**Comment by joebernard** [comment]
I've tried setting `requestIOSPermissions` in both `Amplify.configure` and `PushNotification.configure`. Neither works. I'm prompted to allow notifications when the app starts.

```javascript
Amplify.configure({...myAmplifyConfig})
PushNotification.configure({
  appId: "my app id",
  requestIOSPermissions: false,
})
```

```javascript
Amplify.configure({
  ...myAmplifyConfig,
  PushNotification: {
    requestIOSPermissions: false
  }
})
```

**Comment by amhinson** [comment]
@joebernard Sorry for the trouble!

I'm testing this and have been able to reproduce this behavior. I added some logging and It looks like the `configure` method for `PushNotification` is being called twice. The first time is from `Amplify.configure()` (https://github.com/aws-amplify/amplify-js/blob/master/packages/core/src/Amplify.ts#L43) and the 2nd is from `PushNotification.configure()`.

Since `Amplify.configure` is being called with your base config, `requestIOSPermissions` is defaulting to `true` (https://github.com/aws-amplify/amplify-js/blob/master/packages/pushnotification/src/PushNotification.ts#L79), so when `PushNotification.configure` is called with `requestIOSPermissions` set to `false`, the permission request is already showing.

Could you try:
- Remove `PushNotification.configure` entirely.
- Change `Amplify.configure` to the following:

```js
Amplify.configure({
    ...myAmplifyConfig,
    requestIOSPermissions: false
})
```

Let me know how that turns out. We will be working on updating the documentation and look into make this setup easier to understand.

**Comment by joebernard** [comment]
I tried the config change but it still requests permissions when the app loads.

**Comment by amhinson** [comment]
> I tried the config change but it still requests permissions when the app loads.

Sorry about that! I just submitted a PR to address this issue.

**Comment by joebernard** [comment]
Thanks! Once this PR lands, is the correct way to configure `PushNotifications` with

```javascript
Amplify.configure({
  ...myAmplifyConfig,
  requestIOSPermissions: false
})
```
or should the PushNotification config be broken out to a sub-object?
```javascript
Amplify.configure({
  ...myAmplifyConfig,
  PushNotification: {
    requestIOSPermissions: false
  }
})
```
 or with a separate `PushNotification.configure()`?

**Comment by Ashish-Nanda** [comment]
@joebernard we will update it to be:
```js
Amplify.configure({
  ...myAmplifyConfig,
  PushNotification: {
    requestIOSPermissions: false
  }
})
```

Is that how you were using it before the bug?

**Comment by joebernard** [comment]
Yes, though I was calling it with an `appId` from Pinpoint as well. Here is my complete config.

```javascript
{
  Auth: {
    userPoolId: Config.COGNITO_USER_POOL_ID,
    region: Config.COGNITO_REGION,
    userPoolWebClientId: Config.COGNITO_USER_POOL_WEB_CLIENT_ID,
    identityPoolId: Config.COGNITO_IDENTITY_POOL_ID,
    authenticationFlowType: Config.COGNITO_AUTHENTICATION_FLOW_TYPE,
    oauth: {
      domain: Config.COGNITO_DOMAIN,
      scope: [
        'phone',
        'email',
        'profile',
        'openid',
        'aws.cognito.signin.user.admin',
      ],
      redirectSignIn: 'https://app.mydomain.ai/signIn',
      redirectSignOut: 'https://app.mydomain.ai/signOut',
      responseType: 'code',
    },
  },
  Analytics: {
    AWSPinpoint: {
      appId: Config.PINPOINT_APP_ID,
      region: Config.PINPOINT_REGION,
      resendLimit: 5,
    },
  },
  Storage: {
    AWSS3: {
      bucket: Config.POST_IMAGES_UPLOAD_BUCKET,
      region: Config.S3_REGION,
    },
  },
  PushNotification: {
    appId: Config.PINPOINT_APP_ID,
    requestIOSPermissions: false,
  },
}
```

**Comment by amhinson** [comment]
@joebernard yep the config you posted should work now. Can you test with the `unstable` tag:

`yarn add @aws-amplify/pushnotification@unstable`

**Comment by joebernard** [comment]
It works with my config and the `@unstable` tag.

**Comment by amhinson** [comment]
This fix is now published to `@latest` (3.1.1).
